### DISCUSSION TITLE
Which formulas have you used in simmer.plot for displaying statistics

### DISCUSSION THREAD

**Discussion by ValerCheck** [r-simmer/simmer#280] [General]
Hi there.

I have followed through the simmer.plot code for displaying resources usage or utilization and I'm interested in the formulas you have used in calculations. I couldn't find any references to some scientific papers or books, articles which could shed the light on the usage of implemented formulas :)

For example for usage:
```r
mean = c(0, cumsum(utils::head(.data$value, -1) * diff(.data$time))) / .data$time
```
or for utilization:
```r
utilization = sum(.data$in_use, na.rm = TRUE) / sum(.data$dt, na.rm=TRUE)
```
I am just trying to understand the nature of those values but without some explanation it is a bit complicated. Thanks in advance.

**Comment by Enchufa2** [comment]
> For example for usage:
> 
> ```r
> mean = c(0, cumsum(utils::head(.data$value, -1) * diff(.data$time))) / .data$time
> ```

This is a cumulative weighted mean. For each _t_, we compute the average up to this _t_ weighted by the duration of each observation. 

> or for utilization:
> 
> ```r
> utilization = sum(.data$in_use, na.rm = TRUE) / sum(.data$dt, na.rm=TRUE)
> ```

And this is a simple weighted mean. The quotient `server / capacity` (the _instantaneous_ utilization) is weighted again by the duration of each observation, and averaged. This one is a bit more complicated because, when a resource is closed (`capacity < server`), we take `server` as the capacity (and thus, the utilization is considered to be 100%); and in the special case in which `capacity` is 0, we set `dt = 0` to avoid an `Inf` there.

**Reply by ValerCheck** [reply]
Thanks, will study it further :)
